rootfs/etc/nginx/lua/configuration.lua

```diff
@@ -27,13 +29,61 @@ local function fetch_request_body()
   return body
 end
 
+-- Returns the certificate and key for a given host
+function _M.get_cert_key(hostname)
+  return configuration_data:get(hostname)
+end
+
+local function handle_cert_request()
+  if ngx.var.request_method ~= "POST" then
+    ngx.status = ngx.HTTP_BAD_REQUEST
+    ngx.print("Only POST requests are allowed!")
+    return
+  end
+
+  local cert_str = fetch_request_body()
+
+  local ok, certs = pcall(json.decode, cert_str)
+  if not ok then
+    ngx.log(ngx.ERR,  "could not parse certificate: " .. tostring(certs))
+    return
+  end
+
+  if not certs then
+    ngx.log(ngx.ERR, "certificate dynamic-configuration: unable to read valid request body")
+    ngx.status = ngx.HTTP_BAD_REQUEST
+    return
+  end
+
+  -- Update certificates and private keys for each host
+  for _, cert in pairs(certs) do
+    if cert.hostname and cert.sslCert.pemCertKey then
+      local success, err = configuration_data:set(cert.hostname, cert.sslCert.pemCertKey)
+      if not success then
+        ngx.log(ngx.ERR, "certificate dynamic-configuration: error setting certificate: "
+            .. tostring(err), cert.hostname)
+        ngx.status = ngx.HTTP_BAD_REQUEST
+        return
```

Comment:
You are short circuiting when we fail to store one cert and never try the rest. I suggest we go through the whole list and try to store all of them and collect the errors. In the end check if there was any error and if there was any then return internal server error and log